frontend/dockerfile/docs/reference.md

```diff
@@ -1595,6 +1597,26 @@ See [`COPY --link`](#copy---link).
 
 See [`COPY --exclude`](#copy---exclude).
 
+### ADD --unpack
+
+```dockerfile
+ADD [--unpack=<bool>] <src> ... <dir>
+```
+
+The `--unpack` flag controls whether or not to automatically unpack tar
+archives (including compressed formats like `gzip` or `bzip2`) when adding them
```

Comment:
Could there be a bit more clarity about the compressed format support? Is this format support coming from a Go package or local commands on the build system?

It'd save some guess work, but I guess it's easy enough to try as-is with other formats when encountered to know if that works (eg: `zstd` which is becoming more commonly adopted).